src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -1,68 +1,104 @@
+import { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import {
-  ActionRow, Card, Dropdown, Icon, IconButton,
+  ActionRow, Card, Dropdown, Icon, IconButton, useToggle,
 } from '@edx/paragon';
 import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@edx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { scrollToElement } from '../../course-outline/utils';
 import messages from './messages';
 
-const CourseXBlock = ({ id, title }) => {
+const CourseXBlock = ({
+  id, title, unitXBlockActions, shouldScroll,
+}) => {
+  const currentRef = useRef(null);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
   const intl = useIntl();
 
+  const onXBlockDelete = () => {
+    unitXBlockActions.handleDelete(id);
+    closeDeleteModal();
+  };
+
+  useEffect(() => {
+    // if this item has been newly added, scroll to it.
+    if (currentRef.current && shouldScroll) {
+      scrollToElement(currentRef.current);
+    }
+  }, []);
+
   return (
-    <Card className="mb-1">
-      <Card.Header
-        title={title}
-        actions={(
-          <ActionRow>
-            <IconButton
-              alt={intl.formatMessage(messages.blockAltButtonEdit)}
-              iconAs={EditIcon}
-              size="md"
-              onClick={() => {}}
-            />
-            <Dropdown>
-              <Dropdown.Toggle
-                id={id}
-                as={IconButton}
-                src={MoveVertIcon}
-                alt={intl.formatMessage(messages.blockActionsDropdownAlt)}
-                size="sm"
-                iconAs={Icon}
+    <div ref={currentRef} data-testid="course-xblock">
```

Comment:
If `data-testid` is needed for wrapper element, use 
```
const CourseXBlock = ({
  id, title, unitXBlockActions, shouldScroll, ...props,
}) => {
```
plus

`<div ref={currentRef} data-testid="course-xblock" {...props}>`

Add `data-testid` for elements that are not wrappers

Reply:
Changed